pkg/ccl/workloadccl/workloadccl_test: TestImportFixtureCSVServer failed [integer divide by zero]

pkg/ccl/workloadccl/workloadccl_test.TestImportFixtureCSVServer [failed](https://mesolite.cluster.engflow.com/invocations/default/d64de354-0463-4ccd-8bed-9e5a94b069b8?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3dvcmtsb2FkY2NsOndvcmtsb2FkY2NsX3Rlc3Q=) on master @ [59848e274abde8644a3f005d711e0c5e484b8d20](https://github.com/cockroachdb/cockroach/commits/59848e274abde8644a3f005d711e0c5e484b8d20):

Fatal error:

```
panic: runtime error: integer divide by zero [recovered]
	panic: runtime error: integer divide by zero [recovered]
	panic: runtime error: integer divide by zero [recovered]
	panic: runtime error: integer divide by zero [recovered]
	panic: runtime error: integer divide by zero

```
Stack: 

```
goroutine 3518 [running]:
testing.tRunner.func1.2({0x6506120, 0xca9ba70})
	GOROOT/src/testing/testing.go:1631 +0x24a
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1634 +0x377
panic({0x6506120?, 0xca9ba70?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	pkg/util/leaktest/leaktest.go:132 +0x3aa
panic({0x6506120?, 0xca9ba70?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x49bc65?, {0x893d890, 0xd316600})
	pkg/util/stop/stopper.go:226 +0x65
panic({0x6506120?, 0xca9ba70?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0xc0077a6ea0, {0x893d890, 0xd316600})
	pkg/util/stop/stopper.go:535 +0x170
panic({0x6506120?, 0xca9ba70?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/ccl/workloadccl.csvServerPaths({0xc0021927f8?, 0x0?}, {0x7f638e88d1f8?, 0xc00b1e9100?}, {{0x6fe8b3e, 0x2}, 0x0, {0x70b48d3, 0x20}, {0xa, ...}, ...}, ...)
	pkg/ccl/workloadccl/fixture.go:171 +0x6ba
github.com/cockroachdb/cockroach/pkg/ccl/workloadccl.ImportFixture({0x893d890, 0xd316600}, 0xc00b08d790, {0x7f638e88d1f8, 0xc00b1e9100}, {0x88a65f0, 0x1}, 0x1, 0x0, {0xc0021927f8, ...})
	pkg/ccl/workloadccl/fixture.go:430 +0x6b7
pkg/ccl/workloadccl/workloadccl_test_test.TestImportFixtureCSVServer(0xc004207a00)
	pkg/ccl/workloadccl/fixture_test.go:259 +0x44f
testing.tRunner(0xc004207a00, 0x74d60f0)
	GOROOT/src/testing/testing.go:1689 +0xfb
created by testing.(*T).Run in goroutine 1
	GOROOT/src/testing/testing.go:1742 +0x390
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestImportFixtureCSVServer
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestImportFixtureCSVServer3930782724
    test_log_scope.go:76: use -show-logs to present logs inline
*
* ERROR: a panic has occurred!
* runtime error: integer divide by zero
* (1) attached stack trace
*   -- stack trace:
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop
*   | 	pkg/util/stop/stopper.go:535
*   | runtime.gopanic
*   | 	GOROOT/src/runtime/panic.go:770
*   | runtime.panicdivide
*   | 	GOROOT/src/runtime/panic.go:240
*   | github.com/cockroachdb/cockroach/pkg/ccl/workloadccl.csvServerPaths
*   | 	pkg/ccl/workloadccl/fixture.go:171
*   | github.com/cockroachdb/cockroach/pkg/ccl/workloadccl.ImportFixture
*   | 	pkg/ccl/workloadccl/fixture.go:430
*   | pkg/ccl/workloadccl/workloadccl_test_test.TestImportFixtureCSVServer
*   | 	pkg/ccl/workloadccl/fixture_test.go:259
*   | testing.tRunner
*   | 	GOROOT/src/testing/testing.go:1689
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: integer divide by zero
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
    stopper.go:226: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: outputs.zip/logTestImportFixtureCSVServer3930782724
    stopper.go:226: panic: runtime error: integer divide by zero
--- FAIL: TestImportFixtureCSVServer (3.00s)
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportFixtureCSVServer.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45179

pkg/ccl/workloadccl/workloadccl_test.TestImportFixtureCSVServer [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18032331?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18032331?buildTab=artifacts#/) on master @ [757b9ab8278cbbfb9ce468341e71a3611d7857b2](https://github.com/cockroachdb/cockroach/commits/757b9ab8278cbbfb9ce468341e71a3611d7857b2):


```
*   | 	GOROOT/src/testing/testing.go:1689
*   | runtime.goexit
*   | 	src/runtime/asm_arm64.s:1222
* Wraps: (2) runtime error: integer divide by zero
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
    stopper.go:226: -- test log scope end --

ERROR: a panic has occurred!
Details cannot be printed yet because we are still unwinding.
Hopefully the test harness prints the panic below, otherwise check the test logs.

test logs left over in: /artifacts/tmp/_tmp/980e9bdd417e6f4a8319ecf54422a465/logTestImportFixtureCSVServer1558680646
    stopper.go:226: panic: runtime error: integer divide by zero
--- FAIL: TestImportFixtureCSVServer (0.54s)
panic: runtime error: integer divide by zero [recovered]
	panic: runtime error: integer divide by zero [recovered]
	panic: runtime error: integer divide by zero [recovered]
	panic: runtime error: integer divide by zero [recovered]
	panic: runtime error: integer divide by zero

goroutine 2978 [running]:
testing.tRunner.func1.2({0x5c75180, 0xbeb32d0})
	GOROOT/src/testing/testing.go:1631 +0x1c4
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1634 +0x33c
panic({0x5c75180?, 0xbeb32d0?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	pkg/util/leaktest/leaktest.go:132 +0x384
panic({0x5c75180?, 0xbeb32d0?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x4004f66eb8?, {0x80a5140, 0xc5b3bc0})
	pkg/util/stop/stopper.go:226 +0x68
panic({0x5c75180?, 0xbeb32d0?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Stop(0x40031bbe60, {0x80a5140, 0xc5b3bc0})
	pkg/util/stop/stopper.go:535 +0x154
panic({0x5c75180?, 0xbeb32d0?})
	GOROOT/src/runtime/panic.go:770 +0x124
github.com/cockroachdb/cockroach/pkg/ccl/workloadccl.csvServerPaths({0x40048412c0?, 0x7fffffffffffffff?}, {0xffff5a668ba0?, 0x4008421d00?}, {{0x67575de, 0x2}, 0x0, {0x6824166, 0x20}, {0xa, ...}, ...}, ...)
	pkg/ccl/workloadccl/fixture.go:171 +0x5f8
github.com/cockroachdb/cockroach/pkg/ccl/workloadccl.ImportFixture({0x80a5140, 0xc5b3bc0}, 0x4006eee270, {0xffff5a668ba0, 0x4008421d00}, {0x800d850, 0x1}, 0x1, 0x0, {0x40048412c0, ...})
	pkg/ccl/workloadccl/fixture.go:430 +0x53c
pkg/ccl/workloadccl/workloadccl_test_test.TestImportFixtureCSVServer(0x4005b29380)
	pkg/ccl/workloadccl/fixture_test.go:259 +0x370
testing.tRunner(0x4005b29380, 0x6c44ee8)
	GOROOT/src/testing/testing.go:1689 +0xec
created by testing.(*T).Run in goroutine 1
	GOROOT/src/testing/testing.go:1742 +0x318
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestImportFixtureCSVServer.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/ccl/workloadccl/BUILD.bazel
+++ b/pkg/ccl/workloadccl/BUILD.bazel
@@ -17,7 +17,6 @@ go_library(
         "//pkg/security/username",
         "//pkg/settings/cluster",
         "//pkg/util/ctxgroup",
-        "//pkg/util/errorutil",
         "//pkg/util/humanizeutil",
         "//pkg/util/limit",
         "//pkg/util/log",

--- a/pkg/ccl/workloadccl/fixture.go
+++ b/pkg/ccl/workloadccl/fixture.go
@@ -22,7 +22,6 @@ import (
 	"github.com/cockroachdb/cockroach-go/v2/crdb"
 	"github.com/cockroachdb/cockroach/pkg/cloud"
 	"github.com/cockroachdb/cockroach/pkg/util/ctxgroup"
-	"github.com/cockroachdb/cockroach/pkg/util/errorutil"
 	"github.com/cockroachdb/cockroach/pkg/util/humanizeutil"
 	"github.com/cockroachdb/cockroach/pkg/util/limit"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -328,22 +327,11 @@ func (l ImportDataLoader) InitialDataLoad(
 
 // Specify an explicit empty prefix for crdb_internal to avoid an error if
 // the database we're connected to does not exist.
-const numNodesQuery = `SELECT count(node_id) FROM "".crdb_internal.gossip_liveness`
-const numNodesQuerySQLInstances = `SELECT count(1) FROM system.sql_instances WHERE addr IS NOT NULL`
+const numNodesQuery = `SELECT count(1) FROM system.sql_instances WHERE addr IS NOT NULL`
 
 func getNodeCount(ctx context.Context, sqlDB *gosql.DB) (int, error) {
 	var numNodes int
 	if err := sqlDB.QueryRow(numNodesQuery).Scan(&numNodes); err != nil {
-		// If the query is unsupported because we're in
-		// multi-tenant mode, use the sql_instances table.
-		if !strings.Contains(err.Error(), errorutil.UnsupportedUnderClusterVirtualizationMessage) {
-			return 0, err
-
-		}
-	} else {
-		return numNodes, nil
-	}
-	if err := sqlDB.QueryRow(numNodesQuerySQLInstances).Scan(&numNodes); err != nil {
 		return 0, err
 	}
 	return numNodes, nil

